tracer/src/Datadog.Trace/Debugger/Expressions/ProbeExpressionEvaluator.cs

```diff
@@ -190,6 +206,85 @@ private void EvaluateMetric(ref ExpressionEvaluationResult result, MethodScopeMe
         }
     }
 
+    private void EvaluateSpanDecorations(ref ExpressionEvaluationResult result, MethodScopeMembers scopeMembers)
+    {
+        var resultBuilder = new List<ExpressionEvaluationResult.DecorationResult>();
+        try
+        {
+            EnsureNotNull(_compiledDecorations);
+
+            var compiledDecorations = _compiledDecorations.Value;
+            if (compiledDecorations == null)
+            {
+                Log.Debug($"{nameof(ProbeExpressionEvaluator)}.{nameof(EvaluateSpanDecorations)}: {nameof(_compiledDecorations.Value)} is null");
+                return;
+            }
+
+            for (int i = 0; i < compiledDecorations.Length; i++)
+            {
+                try
+                {
+                    var when = compiledDecorations[i].Key.Delegate(scopeMembers.InvocationTarget, scopeMembers.Return, scopeMembers.Duration, scopeMembers.Exception, scopeMembers.Members);
+                    if (compiledDecorations[i].Key.Errors != null)
+                    {
+                        Log.Debug("{Class}.{Method}: Error when evaluating an expression. {Errors}", nameof(ProbeExpressionEvaluator), nameof(EvaluateSpanDecorations), string.Join(";", compiledDecorations[i].Key.Errors));
+                        (result.Errors ??= new List<EvaluationError>()).AddRange(compiledDecorations[i].Key.Errors);
+                        continue;
+                    }
+
+                    if (!when)
+                    {
+                        Log.Information($"{nameof(ProbeExpressionEvaluator)}.{nameof(EvaluateSpanDecorations)}: Skipping span decoration because the `when` expression was `false`");
+                        continue;
+                    }
+                }
+                catch (Exception e)
+                {
+                    HandleException(ref result, compiledDecorations[i].Key, e);
+                    continue;
+                }
+
+                var compiledExpressions = compiledDecorations[i].Value;
+
+                for (int j = 0; j < compiledExpressions.Length; j++)
+                {
+                    try
+                    {
+                        var currentExpression = SpanDecorations[i].Value[j];
```

Comment:
It seem odd that you're running on `compiledExpressions` and dereferencing `SpanDecorations`. Is it on purpose?

Reply:
You are right. The reason is because I need `DebuggerExpression` and not `CompiledExpression`